packages/yourloops/components/dialogs/medical-record-delete-dialog.tsx

```diff
@@ -0,0 +1,115 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { makeStyles } from "@material-ui/core/styles";
+
+import Button from "@material-ui/core/Button";
+import Dialog from "@material-ui/core/Dialog";
+import DialogActions from "@material-ui/core/DialogActions";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogContentText from "@material-ui/core/DialogContentText";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import Typography from "@material-ui/core/Typography";
+import { makeButtonsStyles } from "../theme";
+import { MedicalRecord } from "../../lib/medical-files/model";
+import MedicalFilesApi from "../../lib/medical-files/medical-files-api";
+import { useAlert } from "../utils/snackbar";
+import ProgressIconButtonWrapper from "../buttons/progress-icon-button-wrapper";
+
+interface Props {
+  onClose: () => void;
+  onDelete: (medicalRecordId: string) => void;
+  medicalRecord: MedicalRecord;
+}
+
+const buttons = makeStyles(makeButtonsStyles);
+
+export default function MedicalRecordDeleteDialog({ onClose, medicalRecord, onDelete }: Props): JSX.Element {
+  const { alertActionButton } = buttons();
+  const { t } = useTranslation("yourloops");
+  const alert = useAlert();
+
+  const [inProgress, setInProgress] = useState<boolean>(false);
+
+  const deleteMedicalRecord = async () => {
+    try {
+      setInProgress(true);
+      await MedicalFilesApi.deleteMedicalRecord(medicalRecord.id);
+      onDelete(medicalRecord.id);
+      setInProgress(false);
+      alert.success(t("medical-record-delete-success"));
+      onClose();
+    } catch (err) {
+      console.log(err);
+      setInProgress(false);
+      alert.error(t("medical-record-delete-failed"));
+    }
+  };
+
+  return (
+    <Dialog
+      open
+      fullWidth
+      maxWidth="sm"
+      onClose={onClose}
+    >
+      <DialogTitle>
+        <Typography variant="h5">
+          {t("delete-medical-report")}
+        </Typography>
+      </DialogTitle>
+
+      <DialogContent>
+        <DialogContentText>
+          {t("delete-warning")} {t("medical-record-pdf")}{new Date(medicalRecord.creationDate).toLocaleDateString()} ?
```

Comment:
This translation key (delete-warning) should handle parameters in order to properly translate the "?" (there is a space before it in French, but not in English